d2/src/main/java/com/linkedin/d2/balancer/strategies/degrader/DegraderLoadBalancerStrategyV3.java

```diff
@@ -931,7 +931,8 @@ private static void enrollNewClientInRecoveryMap(Map<DegraderTrackerClient, Doub
         && !state.getTrackerClients().contains(client)                        // client is new
         && config.getRingRampFactor() > FAST_RECOVERY_THRESHOLD               // Fast recovery is enabled
         && degraderControl.getInitialDropRate() > SLOW_START_THRESHOLD        // Slow start is enabled
-        && !degraderControl.isHigh())                                         // current client is not degrading or QPS is too low
+        && !degraderControl.isHigh()                                          // current client is not degrading or QPS is too low
+        && !client.doNotSlowStart())                                          // doNotSlowStart is set to false
```

Comment:
Any comment on this?

Reply:
I think the current implementation will not enroll new tracker client into fast recovery if doNotSlowStart is enabled. 
https://github.com/linkedin/rest.li/blob/7372bc76f7ce78007ecb24ca80f17399e5246e13/d2/src/main/java/com/linkedin/d2/balancer/clients/DegraderTrackerClientImpl.java#L80
Here, if doNotSlowStart is enabled, initialDropRate will be set to 0, which fails the check on line 933 `degraderControl.getInitialDropRate() > SLOW_START_THRESHOLD`. My change ensures that if we override doNotSlowStart by setDoNotSlowStart(), the same behavior is expected.

Reply:
It works before because when trackerClient set doNotSlowStart to be true, the initialDropRate will always be 0, and now it's not guaranteed? Is that why we need to add this? Or is this a bug before?

Reply:
setDoNotSlowStart() is a new method I exposed with this change. Previously, doNotSlowStart is only determined when creating the TrackerClient with `uriSpecificProperties`. Now, for subsetting, it introduces a new case where a TrackerClient can be created, but add to the load balancer later. So before adding the TrackerClient to the load balancer, we need to manually set doNotSlowStart and make it not enroll in fast recovery program.

Reply:
Do we need to change the same logic in RLB QuarantineManager.java line 380?

Reply:
There are two cases for clients to be put into RecoveryMap:
1. It's a newly added client and slow start is enabled.
2. The client is just evicted from the quarantine.

The case I added above in degrader is for case 1, which is already implemented by RLB: https://github.com/linkedin/rest.li/blob/7372bc76f7ce78007ecb24ca80f17399e5246e13/d2/src/main/java/com/linkedin/d2/balancer/strategies/relative/StateUpdater.java#L317.

For the quarantine case, the doNotSlowStart flag should not have an effect here. In degrader, it is handled by https://github.com/linkedin/rest.li/blob/7372bc76f7ce78007ecb24ca80f17399e5246e13/d2/src/main/java/com/linkedin/d2/balancer/strategies/degrader/DegraderLoadBalancerStrategyV3.java#L665.